frontend/src/metabase/common/hooks/use-setting/use-setting.ts

```diff
@@ -40,3 +50,36 @@ export const useUserSetting = <T extends keyof UserSettings>(
   );
   return [currentValue, shouldDebounce ? debouncedSetter : setter];
 };
+
+export const useMergeSetting = <Key extends SettingKey>(
+  displaySetting: SettingElement<Key>,
+): SettingElement<Key> => {
+  const settingDefinitions = useSelector(getAdminSettingDefinitions);
+  const apiSetting = settingDefinitions.find(
+    (setting: SettingDefinition) => setting.key === displaySetting.key,
+  ) as SettingDefinition<Key> | undefined;
+
+  const mergedSetting: SettingElement<Key> = useMemo(() => {
+    return {
+      ...(apiSetting ?? {}),
+      ...displaySetting,
+    };
+  }, [apiSetting, displaySetting]);
+
+  return mergedSetting;
+};
+
+export const useGetSetSetting = <Key extends SettingKey>(
```

Comment:
as comment somewhere else I don't think we need this anymore since the call site doesn't use the callback.